src/nwb_conversion_tools/basedatainterface.py

```diff
@@ -40,6 +43,31 @@ def get_conversion_options(self):
         return dict()
 
     @abstractmethod
-    def run_conversion(self, nwbfile_path: str, metadata: dict, **conversion_options):
-        """Child DataInterface classes should override this to perform their conversion."""
-        pass
+    def run_conversion(
+        self,
+        nwbfile_path: Optional[str] = None,
+        nwbfile: Optional[NWBFile] = None,
+        metadata: Optional[dict] = None,
+        overwrite: bool = False,
+        **conversion_options,
+    ):
+        """
+        Run the NWB conversion for the instantiated data interface.
+
+        Parameters
+        ----------
+        nwbfile_path: FilePathType
+            Path for where to write or load (if overwrite=False) the NWBFile.
+            If specified, this context will always write to the this location.
+        nwbfile: NWBFile, optional
+            An in-memory NWBFile object to write to the location.
+        metadata: dict, optional
+            Metadata dictionary with information used to create the NWBFile when one does not exist or overwrite=True.
+        overwrite: bool, optional
+            Whether nor not to overwrite the NWBFile if one exists at the nwbfile_path.
+            The default is False (append mode).
+        verbose: bool, optional
+            If 'nwbfile_path' is specified, informs user after a successful write operation.
+            The default is True.
+        """
```

Comment:
> I dislike the verbose true by default.

Then bring it up as a separate issue, this PR and all others are keeping back-compatability as much as possible aside from the depreciated arguments.

My big point to verbosity is as you and I have discussed previously, `tqdm` bars are considered 'verbosity' and they are quite a lovely addition to any conversion involving multiple data sources, multiple data usage stages, or even iterating over multiple converters in parallel. Also, all `dandi` tools are verbose by default, too, and they should be considered close companions much like the NWBInspector.

Reply:
> because it forces us to add code to the examples in the gallery where I think that the aim is that they should be as simple a possible.

I can't even see that stuff with the toggle prompt off, and it seems a very small price to pay

Reply:
I actually didn't understand your argument about `tqdm` until now. You are saying that 1) tqdm is also considered verbosity, 2) we like tqdm, 3) so all verbosity (not just tqdm) should be there.
The ecosystem consistency is a good argument. 

Let's discuss this into another PR as you said, I was just pointing out an instance of this having an unintended consequence that requires expanding code. Similar to what you did into the other #527 just after this discussion.